src/com/projectkorra/projectkorra/waterbending/passive/FastSwim.java

```diff
@@ -39,10 +39,15 @@ public FastSwim(final Player player) {
 
 	@Override
 	public void progress() {
-		if (!this.bPlayer.canUsePassive(this) || !this.bPlayer.canBendPassive(this) || CoreAbility.hasAbility(this.player, WaterSpout.class) || CoreAbility.hasAbility(this.player, EarthArmor.class)) {
+		if (!this.bPlayer.canUsePassive(this) || !this.bPlayer.canBendPassive(this) || CoreAbility.hasAbility(this.player, EarthArmor.class)) {
 			this.remove();
 			return;
 		}
+
+		//Don't remove, for SpoutHop - avoids forcing SpoutHop users to re-sneak and recreate FastSwim (Less clunky)
+		if (CoreAbility.hasAbility(this.player, WaterSpout.class)) {
```

Comment:
Should we have a check here that checks if SpoutHop is disabled, and if so, remove?

Reply:
This seems fine; I have committed this change 👍